lib/util.js

```diff
@@ -501,22 +508,47 @@ module.exports = {
           // TODO: have to add auth here (but first, auth to be put into the openapi tree)
         });
 
+        // If a folder has only one child which is a folder then we collapsed the child folder
+        // with parent folder.
+        /* eslint-disable max-depth */
+        if (resource.childCount === 1) {
+          for (let subChild in resource.children) {
+            let resourceSubChild = resource.children[subChild];
+            if (resourceSubChild.requests.length > 0) {
+              itemGroup.name += '/' + resourceSubChild.name;
+              for (let i = 0; i < resourceSubChild.requests.length; i++) {
+                itemGroup.items.add(
+                  this.convertChildToItemGroup(openapi, resourceSubChild.requests[i])
+                );
+              }
+            }
+            else {
+              resourceSubChild.name = resource.name + '/' + resourceSubChild.name;
+              return this.convertChildToItemGroup(openapi, resourceSubChild);
+            }
+          }
+        }
+        /* eslint-enable */
+        else {
         // recurse over child leaf nodes
         // and add as children to this folder
-        for (i = 0, requestCount = resource.requests.length; i < requestCount; i++) {
-          itemGroup.items.add(
-            this.convertChildToItemGroup(openapi, resource.requests[i])
-          );
-        }
-
-        // recurse over child folders
-        // and add as child folders to this folder
-        for (subChild in resource.children) {
-          if (resource.children.hasOwnProperty(subChild) && resource.children[subChild].requestCount > 0) {
+          for (i = 0, requestCount = resource.requests.length; i < requestCount; i++) {
             itemGroup.items.add(
-              this.convertChildToItemGroup(openapi, resource.children[subChild])
+              this.convertChildToItemGroup(openapi, resource.requests[i])
```

Comment:
Same comment
Can we make a direct call to this.convertRequestToItem(openapi, resourceSubChild.requests[i]) ?
As this.convertChildToItemGroup(openapi, resourceSubChild.requests[i]) just checks whether if it is of type item-group or not. If not, calls this.convertRequestToItem().